runtime/src/main/java/io/quarkiverse/githubaction/runtime/GitHubEvent.java

```diff
@@ -159,34 +159,27 @@ public DynamicGraphQLClient getDynamicGraphQLClient() {
     }
 
     public GHRepository getGHRepository() {
-        var repo = gitHubRepository;
-        if (repo == null) {
+        GHRepository localGitHubRepository = gitHubRepository;
+        if (localGitHubRepository == null) {
             synchronized (this) {
                 if (gitHubRepository == null) {
-                    gitHubRepository = repo = repository();
+                    gitHubRepository = localGitHubRepository = createGHRepository(getGitHub(), context.getGitHubRepository());
                 }
             }
         }
-        return repo;
+        return localGitHubRepository;
     }
 
-    private GHRepository repository() {
-        var gitHub = getGitHub();
-        return gitHub.isOffline() ? offlineRepository() : onlineRepository(gitHub);
-    }
+    private static GHRepository createGHRepository(GitHub gitHub, String repositoryName) {
```

Comment:
Have a look at the result, not at the diff but the idea here is to have the logic in one method. It's probably a matter of personal taste but having the flow split in several methods made it a bit harder to follow IMO.